src/data_structure/temporal_structure.jl

```diff
@@ -700,7 +700,15 @@ function unit_dynamic_time_indices(
     (
         (unit=u, t_before=tb, t_after=ta)
         for (u, blk) in units_on__temporal_block(unit=unit, _compact=false)
-        for (tb, ta) in dynamic_time_indices(m, blk; t_before=t_before, t_after=t_after)
+        for (tb, ta) in dynamic_time_indices(m, blk; t_before=t_before, t_after=t_after) 
+        # When representative temporal structure is used, the dynamic time indices of a unit must stay within the 
+        # represented temporal blocks, including the corresponding representing blocks in case any of them leads to
+        # time slices outside the represented blocks. The reason is that the constraints about unit state transition 
+        # may use variables (e.g. units_on) with time indices only inside the represented temporal blocks, otherwise 
+        # the model can't find a mapping to representative blocks for the outside time indices.
+        if !(explicit_representative_temporal_block(temporal_block=blk)) || tb in time_slice(
+            m; temporal_block=members(blk)
+        )
```

Comment:
This will damage performance.

Reply:
Same concern. I've created issue #1018 to explain the problem. This change attempts for option 1. The latest commit ca43ceb8f8c66611e6d0977e82b189cb45edbbb4 attempts for option 2 and should be more efficient, but it couldn't pass all tests.